tracer/test/Datadog.Trace.Security.Unit.Tests/IAST/Tainted/VulnerabilityBatchTests.cs

```diff
@@ -0,0 +1,105 @@
+// <copyright file="VulnerabilityBatchTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Iast;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Security.Unit.Tests.IAST.Tainted;
+
+public class VulnerabilityBatchTests
+{
+    [Fact]
+    public void GivenAnVulnerabilityBatch_WhenAddVulnerability_VulnerabilityIsCorrectlyAdded()
+    {
+        var batch = new VulnerabilityBatch();
+        var source = new Source(2, "name", "value");
+        var source2 = new Source(3, "name2", "value2");
+        var ranges = new Range[] { new Range(0, 2, source), new Range(2, 2, source2) };
+        var evidence = new Evidence("sql_query", ranges);
+        var vulnerability = new Vulnerability("sqli", new Location(), evidence);
+        batch.Add(vulnerability);
+
+        batch.Sources.Count.Should().Be(2);
+        batch.Sources[0].GetInternalId().Should().Be(0);
+        batch.Sources[1].GetInternalId().Should().Be(1);
+        batch.Vulnerabilities.Count.Should().Be(1);
+        batch.Vulnerabilities[0].Evidence.ValueParts.Count.Should().Be(2);
+        batch.Vulnerabilities[0].Evidence.ValueParts[0].Source.Should().Be(0);
+        batch.Vulnerabilities[0].Evidence.ValueParts[1].Source.Should().Be(1);
+        batch.Vulnerabilities[0].Evidence.ValueParts[0].Value.Should().Be("sq");
+        batch.Vulnerabilities[0].Evidence.ValueParts[1].Value.Should().Be("l_");
```

Comment:
Just FYI, I _think_ you can rewrite this sort of code as:

```csharp
batch.Vulnerabilities.Should().BeEquivalentTo(new [] {
  new {
    Evidence = {
        ValueParts = new [] {
           new { Source = 0, Value = "sq" },
           new { Source = 1, Value = "l_" },
        }
    }
  }
});
```

Not sure if that's easier to read (or that it def works), but something to consider maybe 🙂, especially as you could potentially remove some of the duplication between the methods then 🙂

Reply:
I am not sure... ValueParts is not a property that could be set and relies on a private method  call.

Reply:
The `BeEquivalentTo` method uses structural typing, with anonymous objects, so you're not creating real objects here, just "I expect an object with these properties set".

It's not always worthwhile, but I think it maybe makes it more obvious what your testing in this case?

Reply:
Cool. I have added the tests as suggested and I would say that the tests are more readable. Thanks!